src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -45,15 +53,49 @@ class IdentityWalletConnect extends CardanoPeerConnect {
 
     this.sign = async (
       identifier: string,
-      payload: string
+      payload: string,
+      onSignError?: (error: Error) => void
     ): Promise<string> => {
-      if (this.signerCache.get(identifier) === undefined) {
-        this.signerCache.set(
+      let approved: boolean | undefined = undefined;
+      // Closure that updates approved variable
+      const approvalCallback = (approvalStatus: boolean) => {
+        approved = approvalStatus;
+      };
+      this.eventService.emit<PeerConnectSigningEvent>({
+        type: PeerConnectSigningEventTypes.PeerConnectSign,
+        payload: {
           identifier,
-          await Agent.agent.identifiers.getSigner(identifier)
-        );
+          payload,
+          approvalCallback,
+        },
+      });
+      let timeLeft = 3600000;
+      // Wait until approved is true or false
+      while (approved === undefined) {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        timeLeft -= 1000;
+        if (timeLeft <= 0) {
+          if (typeof onSignError === "function") {
+            onSignError(new Error("Time out"));
+          }
+          throw new Error("Time out");
```

Comment:
This also seems weird to me - so now the error will be thrown in our wallet if the dApp doesn't pass `onSignError`? It doesn't fully make sense.

Reply:
I am trying to throw the error both on the dApp and our wallet